incsrc/Algebra.cpp

```diff
@@ -31,6 +31,21 @@ Vec::~Vec()
 }
 
 Vec::Vec(Vec& vec)
+{
+    size = vec.size;
+    data = new float[size];
+    for(int i=0; i<size; i++)
+        data[i] = vec.data[i]; 
+}
+
+Vec::Vec(Vec&& vec)
+{
+    size = vec.size;
+    data = vec.data;
+      
```

Comment:
add
 vec.data = nullptr;
this will ensure that vec.data is pointed to null (which is not data). This will prevent data to be deleted

vec.data->[DATA]<-this.data
vec destructs, then this.data is dangling pointer
vec.data-> ? <-this.data

therefore make sure it is pointed to null
[DATA]<-this.data
vec.data->[nullptr]
[DATA] is safe now

Reply:
Oh I sure can